packages/widget/src/components/Routes/RoutesExpanded.tsx

```diff
@@ -105,10 +105,17 @@ export const RoutesExpandedElement = () => {
   );
 
   const routeNotFound = !currentRoute && !isLoading && !isFetching && expanded;
-
+  const routesAvailable = (routes && routes?.length > 0) || false;
   const toAddressUnsatisfied = currentRoute && requiredToAddress && !toAddress;
   const allowInteraction = account.isConnected && !toAddressUnsatisfied;
 
+  useEffect(() => {
+    !isLoading &&
+      !isFetching &&
+      isFetched &&
+      emitter.emit(WidgetEvent.RoutesAvailable, routesAvailable);
+  }, [routesAvailable, emitter, isLoading, isFetching, isFetched]);
+
```

Comment:
As we discussed in Discord, it would make more sense to have it fire from the `useRoutes` hooks inside the query. That way we can subscribe to the event no matter which widget variant we use - compact or wide. Also, I'd return the routes array so integrators can also make use of it.

Reply:
Thanks! Good points:
a.) It´s sitting within the useRoutes hook now
b.) It´s returning the routes-array